internal/resultstore/resultstore.go

```diff
@@ -56,6 +60,58 @@ func (rs *ResultStore) generatePath(p Pkg) string {
 	return path
 }
 
+func (rs *ResultStore) SaveWriteBufferZip(ctx context.Context, p Pkg, fileName string, writeBufferPaths []string) error {
+	bkt, err := blob.OpenBucket(ctx, rs.bucket)
+	if err != nil {
+		return err
+	}
+	defer bkt.Close()
+
+	uploadPath := filepath.Join(rs.generatePath(p), fileName+".zip")
+	log.Info("Uploading results",
+		"bucket", rs.bucket,
+		"path", uploadPath)
+
+	w, err := bkt.NewWriter(ctx, uploadPath, nil)
+	if err != nil {
+		return err
+	}
+	defer w.Close()
+
+	zipWriter := zip.NewWriter(w)
+	defer zipWriter.Close()
+
+	for _, filePath := range writeBufferPaths {
+		file, err := os.OpenFile(filePath, os.O_RDWR, 0666)
+		if err != nil {
+			return err
+		}
+		fileContents, err := os.ReadFile(filePath)
+		if err != nil {
+			return err
+		}
+		writeBufferId := utils.GetSHA256Hash(fileContents)
```

Comment:
To avoid reading the file in and computing the sha256 hash we could do some parsing on the paths in writeBufferPaths to get only the file name and not the random values added. For example we can do MkDirTemp("",filename.json*) and then get everything before the .json file, but it's not very robust.